lib/workers/repository/update/branch/index.ts

```diff
@@ -811,13 +819,21 @@ export async function processBranch(
         }
       } else if (config.automerge) {
         logger.debug('PR is configured for automerge');
-        const prAutomergeResult = await checkAutoMerge(pr, config);
-        if (prAutomergeResult?.automerged) {
-          return {
-            branchExists,
-            result: BranchResult.Automerged,
-            commitSha,
-          };
+        if (
+          !commitSha &&
+          !config.artifactErrors?.length &&
+          !userRebaseRequested
```

Comment:
i think commitSha is already set when userRebaseRequested was true, so this would probyböy enough
```suggestion
          !commitSha &&
          !config.artifactErrors?.length
```